casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -548,24 +548,15 @@ object Validate {
       block: BlockMessage,
       dag: BlockDag,
       emptyStateHash: StateHash,
-      runtimeManager: RuntimeManager,
-      knownStateHashesContainer: AtomicSyncVarF[F, Set[StateHash]]
+      runtimeManager: RuntimeManager
   )(implicit scheduler: Scheduler): F[Either[BlockStatus, ValidBlock]] =
     for {
-      maybeStateHash <- knownStateHashesContainer
-                         .modify[Either[BlockException, Option[StateHash]]] { knownStateHashes =>
-                           for {
-                             //invalid blocks return None and don't update the checkpoints
-                             validateBlockCheckpointResult <- InterpreterUtil
-                                                               .validateBlockCheckpoint[F](
-                                                                 block,
-                                                                 dag,
-                                                                 knownStateHashes,
-                                                                 runtimeManager
-                                                               )
-                             (maybeStateHash, updatedknownStateHashes) = validateBlockCheckpointResult
-                           } yield (updatedknownStateHashes, maybeStateHash)
-                         }
+      maybeStateHash <- InterpreterUtil
+                         .validateBlockCheckpoint[F](
+                           block,
+                           dag,
+                           runtimeManager
+                         )
```

Comment:
Note we allow for different blocks to be added in parallel. Are you sure removing the synchronization at this level is safe? If so, we at least should add a test for this.

Reply:
As far as I can tell, the only mutable parameter here (which could in theory break something after removing the synchronization) is `runtimeManager`. However, `runtimeManager` is implemented through `SyncVar`, so I do not think it really affects safeness of the code. I can add a test, sure. What kind of a test do you want to see? One that tries to `Validate.transaction` a bunch of blocks simultaneously?

Reply:
That is a good point. The test shouldn't be necessary then. If you're bored enough, I would rather just see integration tests that will most likely force a bunch of blocks to be validated simultaneously.